tests/LSP6KeyManager/tests/LSP6ControlledToken.test.ts

```diff
@@ -0,0 +1,840 @@
+          "setData(bytes32,bytes)",
+          [key, value]
+        );
+
+        const executePayload = context.keyManager
+          .connect(context.accounts[3])
+          .execute(payload);
+
+        await expect(executePayload)
+          .to.be.revertedWithCustomError(context.keyManager, "NotAuthorised")
+          .withArgs(context.accounts[3].address, "CHANGEPERMISSIONS");
+      });
+
+      after(async () => {
+        const key =
+          ERC725YKeys.LSP6["AddressPermissions:Permissions"] +
+          context.accounts[4].address.substring(2);
+        const value = ARRAY_LENGTH.ZERO;
+        const payload = context.token.interface.encodeFunctionData(
+          "setData(bytes32,bytes)",
+          [key, value]
+        );
+
+        await context.keyManager.connect(context.accounts[0]).execute(payload);
+      });
+    });
+
+    describe("testing SETDATA permission", () => {
+      before(async () => {
+        await addControllerWithPermission(
+          context,
+          context.accounts[4],
+          ARRAY_LENGTH.FIVE,
+          ARRAY_INDEX.FOUR,
+          PERMISSIONS.SETDATA
+        );
+      });
+
+      it("should add the new controller without changing other controllers", async () => {
+        // Check that a new controller was added and other controllers remained intact
+        const keys = [
+          ERC725YKeys.LSP6["AddressPermissions[]"].length,
+          ERC725YKeys.LSP6["AddressPermissions[]"].index + ARRAY_INDEX.ZERO,
+          ERC725YKeys.LSP6["AddressPermissions[]"].index + ARRAY_INDEX.ONE,
+          ERC725YKeys.LSP6["AddressPermissions[]"].index + ARRAY_INDEX.TWO,
+          ERC725YKeys.LSP6["AddressPermissions[]"].index + ARRAY_INDEX.THREE,
+          ERC725YKeys.LSP6["AddressPermissions[]"].index + ARRAY_INDEX.FOUR,
+          ERC725YKeys.LSP6["AddressPermissions:Permissions"] +
+            context.accounts[0].address.substring(2),
+          ERC725YKeys.LSP6["AddressPermissions:Permissions"] +
+            context.accounts[1].address.substring(2),
+          ERC725YKeys.LSP6["AddressPermissions:Permissions"] +
+            context.accounts[2].address.substring(2),
+          ERC725YKeys.LSP6["AddressPermissions:Permissions"] +
+            context.accounts[3].address.substring(2),
+          ERC725YKeys.LSP6["AddressPermissions:Permissions"] +
+            context.accounts[4].address.substring(2),
+        ];
+        const values = [
+          ARRAY_LENGTH.FIVE,
+          context.accounts[0].address,
+          context.accounts[1].address,
+          context.accounts[2].address,
+          context.accounts[3].address,
+          context.accounts[4].address,
+          ALL_PERMISSIONS,
+          PERMISSIONS.CHANGEOWNER,
+          PERMISSIONS.CHANGEPERMISSIONS,
+          PERMISSIONS.ADDPERMISSIONS,
+          PERMISSIONS.SETDATA,
+        ];
+
+        expect(
+          await context.token["getData(bytes32[])"](keys)
+        ).to.be.deep.equal(values);
```

Comment:
Can you check that the other tests don't have the `be` like here, but just `to.deep.equal(...)`

```suggestion
        ).to.deep.equal(values);
```